src/chapter02/exercises/exercise03.lean

```diff
@@ -7,13 +7,31 @@ def rational (x : ℝ) :=
 /-- The product of two rational numbers is always rational. -/
 lemma part_a : ∀ {a b : ℝ}, rational a → rational b → rational (a * b) :=
 begin
-  sorry
+  intros a b,
+  rintro ⟨x1,y1,h1⟩ ⟨x2,y2,h2⟩,
+  use [x1*x2, y1*y2],
+  simp,
+  ring_nf,
+  rw [mul_inv, h1, h2],
+  ring_nf,
```

Comment:
I golfed your proof a bit. First, instead of that `h1` you can use the `rfl` trick; this just completely replaces `a` with x1/y1 so you don't need the `rw h1` in line 15. I think `ring_nf` is just basically applying `div_eq_mul_inv`. The `push_cast` tactic moves all the coercion arrows as tightly to the variables as possible, which I thought made it easier.
```
  rintro a b ⟨x1,y1,rfl⟩ ⟨x2,y2,rfl⟩,
  use [x1*x2, y1*y2],
  push_cast,
  simp only [div_eq_mul_inv, mul_inv],
  ring,
```